Source/cmake/WebKitCommon.cmake

```diff
@@ -258,6 +261,67 @@ if (NOT HAS_RUN_WEBKIT_COMMON)
                     "${CMAKE_SOURCE_DIR}"
                     "${CMAKE_BINARY_DIR}"
         )
+    elseif (CMAKE_EXPORT_COMPILE_COMMANDS)
+        # Create a simple symlink in DeveloperTools/compile_commands.json so that
+        # it can be relied upon regardless of unified or non-unified builds.
+        add_custom_command(
+            OUTPUT ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            DEPENDS ${CMAKE_BINARY_DIR}/compile_commands.json
+            COMMAND ${CMAKE_COMMAND} -E create_symlink
+                    ../compile_commands.json
+                    ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+            VERBATIM
+        )
+        add_custom_target(SymlinkCompileCommandsInDeveloperTools
+            ALL
+            DEPENDS ${CMAKE_BINARY_DIR}/DeveloperTools/compile_commands.json
+        )
+    endif ()
+
+    set(webkitbuild_dir ${CMAKE_SOURCE_DIR}/WebKitBuild)
+    cmake_path(IS_PREFIX webkitbuild_dir ${CMAKE_BINARY_DIR} NORMALIZE clangd_auto_setup_default)
+    set(CLANGD_AUTO_SETUP ${clangd_auto_setup_default} CACHE BOOL
+        "Install a .clangd configuration file and a compile_commands.json symlink "
+        "in the root of the source tree to have out-of-the-box code completion "
+        "in editors.")
```

Comment:
Does clangd work with GCC?

Reply:
clang is the only backend for clangd, however you can use gcc to compile and still use clangd. The compile arguments just may differ a bit leading to inaccurate results.

Reply:
Yes as long as clang can parse all the GCC flags being used. In the case of WebKit, this is the case, I tested it.